src/overlay_94.c

```diff
@@ -0,0 +1,203 @@
+#include "assert.h"
+#include "heap.h"
+#include "party.h"
+#include "pokemon.h"
+#include "constants/pokemon.h"
+#include "constants/species.h"
+#include "constants/sndseq.h"
+#include "overlay_94.h"
+
+void ov94_021E5900(struct UnkStruct_Overlay_94_A* unkPtr) {
+    if (unkPtr->unkc80 != NULL) {
+        GF_ASSERT(FALSE);
+    }
+    unkPtr->unkc80 = AllocFromHeap(12, sizeof(struct UnkStruct_Overlay_94_C));
+    MI_CpuClear8(unkPtr->unkc80, sizeof(struct UnkStruct_Overlay_94_C));
+    unkPtr->unkc80->partyMonIndex = unkPtr->partyMonIndex;
+}
+
+BOOL ov94_021E593C(struct UnkStruct_Overlay_94_A* unkPtr) { 
+    struct UnkStruct_Overlay_94_C* unkA = unkPtr->unkc80; 
+    POKEMON* pokemon = GetPartyMonByIndex(unkPtr->unk654->party, unkPtr->partyMonIndex);
+
+    switch (unkA->unk0) {
+    case 0:
+        unkA->species = GetMonData(pokemon, MON_DATA_SPECIES, 0);
+        switch (unkA->species) {
+        case SPECIES_GIRATINA:
+            Mon_UpdateGiratinaForme(pokemon);
+            unkA->unk8 = 65;
+            unkA->unk10 = 0;
+            break;
+        case SPECIES_SHAYMIN:
+            Mon_UpdateShayminForme(pokemon, SHAYMIN_SKY);
+            unkA->unk8 = 35;
+            unkA->unk10 = 1;
+            break;
+        case SPECIES_ROTOM:
+        default:
+            GF_ASSERT(FALSE);
+            break;
+        } 
+        Pokedex_SetMonCaughtFlag(Sav2_Pokedex_get(Fsys_GetSaveDataPtr(unkPtr->unk654->fsys)), pokemon);
+        unkA->unk0++;
+        break;
+    case 1:
+    case 2:
+        unkA->unk0++;
+        break;
+    case 3:
+        ov94_021E5B04(unkPtr);
+        unkA->unk0++;
+        break;
+    case 4:
+        GX_EngineAToggleLayers(1, 1);
+        ov94_021E5BA0(unkA);
+        unkA->unk0++;
+        break;
+    case 5:
+        unkA->unk4++;
+        if(unkA->unk4 == unkA->unk8) {
+            sub_0207ECE0(unkPtr, unkPtr->partyMonIndex);
+        }
+        ov94_021E5C60();
+        if(unkA->unk4 > unkA->unk8) {
+            if(sub_020154B0(unkA->unk18) == 0) {
+                unkA->unk0++;
+            }
+        }
+        break;
+    case 6:
+        ov94_021E5B30(unkPtr);
+        unkA->unk0++;
+        break;
+    case 7:
+        sub_020720D4(pokemon);
+        unkA->unk0++;
+        break;
+    case 8:
+        if(IsCryFinished() == FALSE) {
+            unkA->unk0++;
+        }
+        break;
+    case 9:
+        STRING* str = NewString_ReadMsgData(unkPtr->msgData, 188);
```

Comment:
Doing a little research, it appears the message data referenced here is from `files/msgdata/msg/msg_0300.gmm`, so the `188` here is `msg_0300_00188` from `#include "msgdata/msg/msg_0300.h"`.

Reply:
If that's true, then Overlay 94 should be handling every time the player uses an item on a pokemon/selects them for the battle tower/online battle in the party menu. That would explain why Shaymin and Giratina's form changes are in here. I'll migrate all the functions to party_menu.h/.c and start naming them, though I'm not sure what y'alls naming style/convention is, so let me know if you have any

Reply:
No, don't do that.

Reply:
oh alright!